### PR TITLE
Use ruff for formatting / linting

### PR DESCRIPTION
**Pull request by chrisoro**
:)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,2 +1,51 @@
...
 line-length = 140
+target-version = "py312"
+
+[tool.ruff.format]
+docstring-code-format = true
+docstring-code-line-length = "dynamic"
+indent-style = "space"
+line-ending = "auto"
+quote-style = "double"
+skip-magic-trailing-comma = true
```
Comment (johannes-do) [inline pyproject.toml]: I would like to have the option to decide by the trailing comma if I want to split things in seperate lines or not. Setting this to false and split-on-trailing-comma = true does the trick.

File: pyproject.toml [hunk 2]
```diff
@@ -1,2 +1,51 @@
...
+    "W",
+    "YTT",
+    # "PL",
+    # "PTH",
+    # "RUF",
+]
+ignore = ["E501"]
+
+[tool.ruff.lint.isort]
+split-on-trailing-comma = false
```
Comment (johannes-do) [inline pyproject.toml]: Set to true for reason stated in L12

File: src/item/descr/texture.py [hunk 3]
```diff
@@ -44,14 +41,7 @@ def _find_bullets(
...
     )
-    small_bullets = search(
-        ref=template_list,
-        inp_img=img_item_descr,
-        threshold=threshold,
-        roi=roi_bullets,
-        use_grayscale=True,
-        mode=mode,
-    )
+    small_bullets = search(ref=template_list, inp_img=img_item_descr, threshold=threshold, roi=roi_bullets, use_grayscale=True, mode=mode)
```
Comment (johannes-do) [inline src/item/descr/texture.py]: revert to split by trailing comma

**Review by aeon0 — CHANGES_REQUESTED** [review]
I am convinced. This is great!

**Pull request merged**